hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/ReplicatedMapProxy.java

```diff
@@ -120,18 +120,38 @@ public void initialize() {
         }
         fireMapDataLoadingTasks();
         if (!config.isAsyncFillup()) {
-            for (int i = 0; i < nodeEngine.getPartitionService().getPartitionCount(); i++) {
-                ReplicatedRecordStore store = service.getReplicatedRecordStore(name, false, i);
-                while (store == null || !store.isLoaded()) {
-                    if ((retryCount++) % RETRY_INTERVAL_COUNT == 0) {
-                        requestDataForPartition(i);
-                    }
-                    sleep();
-                    if (store == null) {
-                        store = service.getReplicatedRecordStore(name, false, i);
+            syncFill();
+        }
+    }
+
+    private void syncFill() {
+        int partitionCount = nodeEngine.getPartitionService().getPartitionCount();
+        BitSet nonLoadedStores = new BitSet(partitionCount);
+        int[] retryCount = new int[partitionCount];
+        for (int i = 0; i < partitionCount; i++) {
```

Comment:
Minor: you can use `nonLoadedStores.set(0, partitionCount);` instead of looping.

Reply:
I think even better option is possible :-) - I can renamed the set to loadedStore and mark a bit as set when a partition loaded is completed. In this case the init logic is not required.

Reply:
I tried doing something similar but beware in that the `nextClearBit` will never return -1 but might instead return a number which is even out of the partition ID range. So if all partitions are loaded, `loadedPartitions.nextClearBit()` will return `partitionCount + 1`